nixos/tests/promscale.nix

```diff
@@ -0,0 +1,146 @@
+      db-password = "snakeOilPassword";
+      db-port = 5433;
+      db-ssl-mode = "prefer";
+      db-user = "postgres";
+      labels-cache-size = 1234;
+      log-level = "info";
+      metrics-cache-size = 1234;
+      migrate = "true";
+      scheduled-election-interval = "50s";
+      tput-report = 1234;
+      leader-election-pg-advisory-lock-id = 2;
+      leader-election-pg-advisory-lock-prometheus-timeout = "60s";
+      # BROKEN: If leader-election is set, database command line args are neglected, only envvars are in effect.
+      leader-election-rest = false;
+
+    in
+
+      {
+
+        name = "promscale";
+
+        meta.maintainers = with lib.maintainers; [ _0x4A6F ];
+
+        nodes = {
+
+          postgresql_11 = {
+            services.promscale.enable = true;
+            services.promscale.asyncAcks = async-acks;
+            services.promscale.installTimescaledb = install-timescaledb;
+            services.promscale.useSchemaVersionLease = use-schema-version-lease;
+            services.promscale.web.listenAddress = listen-address;
+            services.promscale.web.listenPort = listen-port;
+            services.promscale.web.telemetryPath = web-telemetry-path;
+            services.promscale.database.connectRetries = db-connect-retries;
+            services.promscale.database.connectionsMax = db-connections-max;
+            services.promscale.database.host = db-host;
+            services.promscale.database.name = db-name;
+            services.promscale.database.passwordFile = "${pkgs.writeText "promscale.db-passwordFile" db-password}";
+            services.promscale.database.port = db-port;
+            services.promscale.database.sslMode = db-ssl-mode;
+            services.promscale.database.user = db-user;
+            services.promscale.labelsCacheSize = labels-cache-size;
+            services.promscale.logLevel = log-level;
+            services.promscale.metricsCacheSize = metrics-cache-size;
+            services.promscale.migrate = migrate;
+            services.promscale.tputReport = tput-report;
+            #services.promscale.leaderElection.pg.advisoryLockId = leader-election-pg-advisory-lock-id;
+            #services.promscale.leaderElection.pg.advisoryLockPrometheusTimeout = leader-election-pg-advisory-lock-prometheus-timeout;
+            services.promscale.leaderElection.rest = leader-election-rest;
+            #services.promscale.leaderElection.scheduledElectionInterval = scheduled-election-interval;
+            # for future command line args
+            services.promscale.extraOptions = [
+              "-db-writer-connection-concurrency 1"
+            # "-web-cors-origin .*" # corsOrigin:0xc0002da140
+            # "-web-cors-origin example.com.*" # corsOrigin:0xc00016df40
+            # "-web-cors-origin example.com/public/.*" # corsOrigin:0xc00016dea0
+            ];
+            services.postgresql = {
+              authentication = pkgs.lib.mkOverride 10 ''
```

Comment:
You can probably get rid of this by using `/run/postgresql/` as "host" in this test. By default postgresql uses the unix user that connects (on NixOS)

Reply:
Doesn't seem to work without `authentication` set and `db-host = "/run/postgresql/";`.
Seems with these settings promscale is started before postgres is up.